applications/NurbsBrepApplication/custom_utilities/NodeCurveGeometry3D.h

```diff
@@ -0,0 +1,135 @@
+#if !defined(KRATOS_NODE_CURVE_GEOMETRY_3D_H_INCLUDED)
+#define KRATOS_NODE_CURVE_GEOMETRY_3D_H_INCLUDED
+
+// System includes
+
+// External includes
+#include <ANurbs/Core>
+
+// Project includes
+#include "includes/define.h"
+#include "includes/Node.h"
+#include "includes/variables.h"
+
+namespace Kratos {
+
+class NodeCurveGeometry3D
+: public ANurbs::CurveGeometryBase<double, Kratos::array_1d<double, 3>>
+{
+protected:
+    using NodePointer = typename Node<3>::Pointer;
+
+public:
+    using CurveGeometryBaseType = ANurbs::CurveGeometryBase<double,
+        Kratos::array_1d<double, 3>>;
+    using typename CurveGeometryBaseType::KnotsType;
+    using typename CurveGeometryBaseType::ScalarType;
+    using typename CurveGeometryBaseType::VectorType;
+ 
+protected:
+    std::vector<NodePointer> mNodes;
+ 
+public:
+    NodeCurveGeometry3D(
+        const int Degree,
+        const int NumberOfNodes)
+        : CurveGeometryBaseType(Degree, NumberOfNodes)
+        , mNodes(NumberOfNodes)
+    {
+    }
+ 
+    NodePointer
+    Node(
+        const int Index
+    ) const
+    {
+        return mNodes[Index];
```

Comment:
I recommend adding debug errors here (and at other places too):
`KRATOS_DEBUG_ERROR_IF(Index >= mNodes.size()) << "..." << std::endl;`

Reply:
👍 Also on my TODO for the library...
Do you think it would be better to throw an error also in release mode?

Reply:
Btw: How can I check inside python if Kratos was compiled in debug mode?